stock_picking_multiple_carrier/stock_picking.py

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
+# © 2016 initOS GmbH
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import models, fields, api
+
+
+class StockPickingDelivery(models.Model):
+
+    _name = 'stock.picking.delivery'
+
+    carrier_id = fields.Many2one(
+        "delivery.carrier",
+        string="Carrier",)
+    carrier_tracking_ref = fields.Char(
+        string="Carrier Tracking Ref",
+        size=32,
+        required=True,)
+    picking_id = fields.Many2one(
+        comodel_name="stock.picking",
+        string="Picking",
+        required=True)
+
+
+@api.model
+def _carrier_tracking_ref_display(self):
+    # use this function with the functional field "carrier_tracking_ref"
+    # in StockPicking class
+    # to show the carrier information in another module
+    # (in sale order for example)
+        deliveries = dict([(picking.id, picking.delivery_ids)
+                           for picking in self])
+        result = {}
+        for record_id in self:
+            refs = map(lambda delivery:
+                       (delivery.carrier_id.name or '') + ': ' +
+                       (delivery.carrier_tracking_ref or ''),
+                       deliveries[record_id])
```

Comment:
There is an error in this line of code. May you please take a look? 
In my case:
KeyError: stock.picking(1,)